### PR TITLE
[#7191] Add Rocky Linux 9 to GitHub Actions workflows (main)

### PR DESCRIPTION
**Pull request by alanking**
In service of #7191 

Due to the differences, I had to split out EL8 and EL9 workflow files.

Please pay special attention to the section about configuring the dnf plugins and enabling the iRODS unstable repo.

Also, I added the workaround described in #7349 to get this to work. If this is not satisfactory as a solution, then this is blocked by #7349 (and #7191 is blocked by extension).

### CODE REVIEW

File: cmake/packaging.cmake
Comment (korydraughn) [inline cmake/packaging.cmake:187]: Why not `EQUALS 7`?

Just curious.

Comment (alanking) [reply cmake/packaging.cmake:187]: This was working for 8, so wanted to box in the effect. I'm okay with either.

Comment (korydraughn) [reply cmake/packaging.cmake:187]: If we can be explicit, let's do that. Just trying to guard against potentially hidden issues in the future.

Comment (alanking) [reply cmake/packaging.cmake:187]: Gosh, I didn't even notice the bit about "centos". Yes, definitely this should be `EQUALS 7`.

**Review by korydraughn — APPROVED** [review]
Looks good to me.

Let's wait for one more approval from @SwooshyCueb before merging.

**Review by SwooshyCueb — APPROVED** [review]

**Pull request merged**
